quinn-h3/src/client.rs

```diff
@@ -318,4 +318,8 @@ impl RecvResponse {
     pub fn body_stream(self) -> RecvBodyStream {
         RecvBodyStream::new(self.recv, self.conn, self.stream_id)
     }
+
+    pub fn body_reader(self) -> RecvBodyReader {
```

Comment:
So can you give me some context on what the difference between `body_stream()` and `body_reader()` is? (Ideally, write the docstring for `body_reader()` -- and maybe also `body_stream()` so we have that documentation, which we're going to need anyway for this public API?)

Reply:
`RecvBodyReader` is an object that implements `AsyncRead`, whereas `RecvBodyStream` can only poll `HttpFrame::Data` by `HttpFrame::Data`. They are not implemented on the same struct because an `io::Read` implementation needs to have an intermediate buffer, in case the slice passed to `read()` is smaller than the received data frame. 

Maybe some code can be dedpuplicated between those two... actually.

Reply:
And maybe `Recv` and `Reader` are redundant in the name.

Reply:
So what is the use case for polling `HttpFrame::Data` by `HttpFrame::Data` vs the use case for `RecvBodyReader`? I do think `BodyReader` is probably a better name.

Reply:
The intention is to offer a similar API to most `tokio`-aware libs. 

`BodyReader` is here to make it possible to use tokio convenience functions, as `tokio::io::read_to_end()` or `tokio::io::copy()`.

`RecvBodyStream` offers a `Stream` implementation on the data, and is the minimal copy-overhead API. Though it will probably change to be able to yield only parts of big `HttpFrame::Data` in the future.